exercises/plot_dets_filesize.py

```diff
@@ -0,0 +1,132 @@
+from databroker import Broker
+import pandas as pd
+
+import os
+
+import datetime
+import time
+from time import mktime
+
+from eiger_io.fs_handler import EigerHandler
+from databroker.assets.handlers import AreaDetectorTiffHandler
+
+import matplotlib.pyplot as plt
+import matplotlib.dates as mdates
+import matplotlib.ticker as ticker
+
+from collections import defaultdict
+
+def find_keys(hdrs, db):
+    '''
+        This function searches for keys that are stored via filestore in a
+        database, and gathers the SPEC id's from them.
+    '''
+    FILESTORE_KEY = "FILESTORE:"
+    keys_dict = defaultdict(lambda : int(0))
+    files = []
+    for hdr in hdrs:
+        for stream_name in hdr.stream_names:
+            events = hdr.events(stream_name=stream_name)
+            events = iter(events)
+            while True:
+                try:
+                    event = next(events)
+                    if "filled" in event:
+                        # there are keys that may not be filled
+                        for key, val in event['filled'].items():
+                            if key not in keys_dict and not val:
```

Comment:
@elParcero ah here's the problem! you're only adding one case and ignoring the rest!
should be:
`if not val:`

Reply:
Thank you! So far it is working, I encountered one error but placed a try except, the error was CursorNotFound error. 

Now if I am able to get this, the next part would be plotting file size vs hour (for both plan_name + detector) this is what I had in mind
```py
plan_names = ['count', 'scan', 'rel_scan']
detectors = ['eiger1m_single_image', 'eiger4m_single_image', 'xray_eye2_image']
f_sizes = dict()

for plan in plan_names:
    for detector in detectors:
        hdrs = iter(db(since='2015-0-01', until='2018-12-31', plan_name = plan) )
        f_sizes[plan + ":" + detector] = file_sizes(hdrs, db, detector)
```
and then in the function file_sizes, it would take in three arguments ```py def file_sizes(hdrs, db, detector):```, and where it says ```py if key in event['data']:``` inside the function, right after that, could I put, ```py if key == detector: ``` which would run the code inside and find file size for that specific plan_name and detector.